docs/GettingStartedDocs/VisualStudioLinux.md

```diff
@@ -0,0 +1,152 @@
+   is required even if the correct value is shown by default. In other words, make sure the
+   connection is shown in **bold**.)
+3. If using Visual Studio 2019, also update Configuration Properties -> Debugging -> Remote
+   Debug Machine to your build machine, again due to a current Visual Studio 2019 bug.
+   (This step is not needed on Visual Studio 2017.)  At this point, you should be able to
+   build and debug the Hello World application. For further discussion, see the
+   [Linux debugging walkthrough](https://docs.microsoft.com/en-us/cpp/linux/deploy-run-and-debug-your-linux-project?view=vs-2019).
+4. Create an enclave library project by right clicking on the solution in the Solution Explorer
+   and selecting Add -> New Project -> Open Enclave TEE Project (Linux).  (If it is not
+   immediately visible, look under Installed -> Visual C++ -> Cross Platform -> Linux.)  Give it a name,
+   LinuxEnclave for example.  This will create a sample enclave with an ecall\_DoWorkInEnclave()
+   method exposed to applications, that will simply call an ocall\_DoWorkInHost() method that
+   will be implemented in the application.   In this walkthrough, we'll leave this project
+   as is for now, but afterwards you can modify it as you like.
+5. Configure the enclave project to use your Linux build environment, as you did in step 2.
+   At this point, the enclave would build, but cannot be run as the application doesn't
+   invoke it yet.
+6. Import the enclave into your application project, by right clicking on the application
+   project in the Solution Explorer and selecting Open Enclave Configuration -> Import Enclave,
+   then navigate to and select the EDL file (_YourEnclaveProjectName_.edl) in your enclave project.
+   This step will modify your application project settings and add some additional files to it,
+   including a C file named _YourEnclaveProjectName_\_host.c.  This C file contains a
+   sample\_enclave\_call() method that will load and call
+   ecall\_DoWorkInEnclave(), and also contains a sample implementation of a ocall\_DoWorkInHost()
+   method that just prints a message when called.  Although the app could be compiled and run
+   at this point, sample\_enclave\_call() is still not called from anywhere.
+7. Open the application's main.cpp (or if you are starting from another existing application,
+   whatever file you want to invoke enclave code from), and add a call to sample\_enclave\_call().
+   For example, update the main.cpp file to look like this, where the extern C declaration is needed
+   because main.cpp is a C++ file whereas the _YourEnclaveProjectName_\_host.c file is a C file:
+```C
+#include <cstdio>
+
+extern "C" {
+    void sample_enclave_call(void);
+};
+
+int main()
+{
+    printf("hello from LinuxApp!\n");
+    sample_enclave_call();
+    return 0;
+}
+```
+8. You can now set breakpoints in Visual Studio, e.g., inside ecall\_DoWorkInEnclave() and inside
```

Comment:
This works nicely.

NTH: In future it would be nice if the user can also debug into packages that are part of OE SDK. This includes libc, libcxx, mbedtls etc.